crates/turbo-tasks/src/value_type.rs

```diff
@@ -51,6 +51,41 @@ pub trait ValueTraitVc:
     fn get_trait_type_id() -> TraitTypeId;
 }
 
+pub trait IntoSuperTrait<T>: ValueTraitVc
```

Comment:
here are the definitions for the new traits

Reply:
Why ain't the normal From and Into traits working for that?

Reply:
FWIW I like having the separation here. The semantics are clearer.

Reply:
I think the reason the `default fn` was needed because non trait types already implement `Into<TraitVc>` which was the reason it never complained about missing functions